source/administration/data-retention.rst

```diff
@@ -0,0 +1,68 @@
+--------------
+
+To set a custom data retention policy:
+
+1. Go to **System Console > Advanced > Data Retention Policy (Beta)**
+2. Set your desired retention policy for messages. Messages, including file attachments, older than the duration you set will be deleted nightly. The minimum message retention time is one day.
+3. Similarly, set your desired retention policy for file uploads. File uploads older than the duration you set will be deleted nightly from your file storage system, either from your local disk or your Amazon S3 service as specified in **System Console > Files > Storage**. The minimum file retention time is one day.
+4. Set the start time of the daily scheduled data retention job. Choose a time when fewer people are using your system. Must be a 24-hour time stamp in the form HH:MM.
+
+Save the settings and restart your server. You're now all set! Data older than the duration you set are now deleted daily at the specified server time.
+
+You may also run the deletion job manually anytime by clicking **Run Deletion Job Now** button in **System Console > Advanced > Data Retention Policy (Beta)**.
+
+Frequently Asked Questions (FAQ)
+---------------------------------
+
+What happens when a message is deleted?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The message is removed from the Mattermost user interface and deleted from the Posts table. The message is no longer searchable and cannot be retrieved in pinned posts or flagged posts lists.
+
+Replies that did not exceed the message duration are still displayed in the user interface. However, further replies to thread are no longer permitted.
+
+What happens when a file is deleted?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The file attachment is removed from the Mattermost user interface, and deleted from the FileInfo table and from your local disk or Amazon S3 service as specified in **System Console > Files > Storage**.
+
+Why didn't an old file get deleted after running the deletion job?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, check the data deletion was successful in the deletion job table in **System Console > Advanced > Data Retention Policy (Beta)**.
+
+Second, if the files were uploaded prior to Mattermost v4.2, you will need to delete the files manually from your local file storage or Amazon S3 storage. You may do this by simply deleting the relevant folders. // XXX George, any other info/help we can provide?
+
+Note that these files will still be removed from the Mattermost messages if they were uploaded in Mattermost v3.5 or later, which contain the `FileInfo table <https://docs.mattermost.com/administration/changelog.html#database-changes-from-v3-4-to-v3-5>`_. 
+
+Why do I see ``Pending`` in the deletion job table with no details?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This means you haven't restarted your server after enabling the data retention policy. Restart your server and try again.
+
```

Comment:
Only if there's no "in progress" data retention job AND they stay pending for more than ~1 minute. Otherwise, they might just be waiting for a slot to run the job in to become available.